contexts/PaymentContext/src/Domain/ChecksumGenerator.php

```diff
@@ -39,7 +36,7 @@ public function createChecksum( string $string ): string {
 	}
 
 	private function normalizeString( string $string ): string {
-		return strtoupper( preg_replace( $this->cleanupPattern, '', $string ) );
+		return strtoupper( str_replace( [ '-', '_', ' ' ], [ '', '', '' ], $string ) );
```

Comment:
I'm torn about this, do not believe in blacklists.
[Your comment](https://github.com/wmde/FundraisingFrontend/pull/940#issuecomment-330871351) is right in that the 0-pad example feels dodgy, but if we keep in mind that the resulting normalized string will "only" be used for checksum generation and never output, the benefits of ignoring all possible funky delimiters outweighs the risk of a unintuitive checksum collision (which is not even a thing because it does not come with the pretense of uniqueness).

Reply:
I mainly went with this approach because it was easier and is simpler than the regex. Try seeing it as a whitelist of characters with special handling, no blacklisting needed :)